### PR TITLE
useChartData and i18n bug fixes

### LINKED ISSUE #586: Sentry: Production error: useChartData
https://sentry.io/organizations/recidiviz-inc/issues/1990286496/?project=5385222&query=is%3Aunresolved

### LINKED ISSUE #587: Sentry: Production error: i18n Object doesn't support property or method 'existsSync'
https://sentry.io/organizations/recidiviz-inc/issues/1990286496/?project=5385222&query=is%3Aunresolved

### PR DESCRIPTION
**Pull request by jovergaag**
## Description of the change
We experienced two bugs today in production. Neither of these errors came up in our QA (and I was not able to reproduce in production based on error stacktrace or fullstory clicks, but I am quite sure the errors occurred because of low quality Internet: error was result of fetching data error,  and host memory: error was result of timing issue.

1. [Sentry issue:](https://sentry.io/organizations/recidiviz-inc/issues/1985175196/?project=5385222&query=is%3Aunresolved) An error with `useChartData` where we were catching a fetch error and logging it, but returning the wrong value type to the component. An empty object was being returned and then we tried to filter that object which produced an error. Changed the value returned to the components and added tests for `useChartData`.

2. [Sentry issue:](https://sentry.io/organizations/recidiviz-inc/issues/1990286496/?project=5385222&query=is%3Aunresolved) An error with i18n where the translate locale was not set before the first `translate` call was made. Locally I was able to reproduce the error by removing the call to `setTranslateLocale` in the `StateCodeContext`. Since that call to `setTranslateLocale` was dependent on `currentStateCode` being set, my guess is that there was a timing issue that `currentStateCode` had not yet been set by the time the first component called `translate`. I moved the `setTranslateLocale` to be in the layout instead of the context to ensure this is never the case. Also added tests.

## Type of change

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

Closes #586 
Closes #587

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: src/api/metrics/metricsClient.js [hunk 1]
```diff
@@ -44,33 +43,6 @@ async function callMetricsApi(endpoint, getTokenSilently) {
   }
 }
 
-/**
- * A synchronous function that fetches data for the given state, metric type, and file. Takes in
- * functions for setting the API response and a flag indicating whether or not we are still
- * awaiting the API response.
- */
-const fetchChartData = async (
```
Comment (jessex) [inline src/api/metrics/metricsClient.js:52]: Detritus that was no longer in use?

File: src/hooks/useChartData.js [hunk 2]
```diff
@@ -31,7 +31,7 @@ import { callMetricsApi, awaitingResults } from "../api/metrics/metricsClient";
  */
 function useChartData(url, file) {
   const { loading, user, getTokenSilently } = useAuth0();
-  const [apiData, setApiData] = useState({});
+  const [apiData, setApiData] = useState([]);
```
Comment (jessex) [inline src/hooks/useChartData.js:34]: Oh, looking closely below, are we sometimes returning an object and sometimes returning an array? Or is it always an array?

**Review by jessex — COMMENTED** [review]
Great test coverage! A couple of questions.

File: src/hooks/useChartData.js (hunk 2 above)
Comment (jovergaag) [reply src/hooks/useChartData.js:34]: `apiData` will always be an array. `useChartData` returns an object that contains `apiData`.

On line 46 and 52, `metricFile` is an array of objects, so confirming `apiData` is always an array.

File: src/api/metrics/metricsClient.js (hunk 1 above)
Comment (jovergaag) [reply src/api/metrics/metricsClient.js:52]: Yes, my guess that it was used at one point, but we ended up moving it into `useChartData` once the fetch was centralized in that hook.

**Review by jessex — APPROVED** [review]
Good to go, pending the build!

**Comment by jovergaag** [comment]
Update on this - I have been QAing this in staging and managed to reproduce one of the errors. The issue cause the original  error that was logged has been fixed, but it seems `logger.error(error);` is actually throwing an error. So by deleting one of the metrics files and forcing another fetch error - I'm still seeing sentry errors. 

The good thing is that they are being handled nicely and just showing our error message instead of crashing the page.

File: src/hooks/useChartData.js [hunk 3]
```diff
@@ -55,7 +54,7 @@ function useChartData(url, file) {
     } catch (error) {
       setAwaitingApi(false);
       setIsError(true);
-      logger.error(error);
+      console.error(error);
```
Comment (jovergaag) [inline src/hooks/useChartData.js:57]: This is an insane IE11 issue. It was throwing an error on `logger.error`: `TypeError: Invalid calling object`. I confirmed this by forcing a fetch error by deleting a metric file in the staging API. Noticed that this error shows up in Sentry, however, there are no errors in the console in Browserstack. After deploying this commit to staging and forcing the error again, I do not see this error in Sentry. So, just removing logger altogether.

Comment (jessex) [reply src/hooks/useChartData.js:57]: Well, that's bizarre...

**Review by jessex — APPROVED** [review]

**Pull request merged**